share/wake/lib/system/sources.wake

```diff
@@ -38,27 +38,38 @@ def raw_source file =
     def get_modtime file = prim "get_modtime"
     def time = get_modtime file
     if time == -1 then
-        makeBadPath (makeError "{file}: source does not exist. Did you delete it or move it without telling git?")
+        failWithError "{file}: source does not exist. Did you delete it or move it without telling git?"
     else
         makeExecPlan ("<source>", str time, file, Nil) Nil
         | setPlanShare       False
         | setPlanEcho        logDebug
         | setPlanEnvironment Nil
         | setPlanFnOutputs   (file, _)
         | runJobWith virtualRunner
-        | getJobOutput
+        | getJobResultOutput
 
 export def source (file: String): Path =
+    sourceResult file
+    | pathResultShim
+
+export def sourceResult (file: String): Result Path Error =
     def basename = replace `^.*/` '' file
     def dirname = simplify "{file}/.."
     match (sources dirname basename.quote)
-        Nil = makeBadPath "{file}: not a source file".makeError
-        x, Nil = x
+        Nil = failWithError "{file}: not a source file"
+        x, Nil = Pass x
         _ = unreachable "a quoted RegExp can never match more than one distinct String"
 
 # Find sources files
 export def sources (dir: String) (filterRegexp: RegExp): List Path =
+    sourcesResult dir filterRegexp
+    | pathListResultShim
```

Comment:
This still removes Failure instead of BadPath..

Reply:
I thought I checked that, but apparently I was just making bad assumptions.  Sorry!